CMakeLists.txt

```diff
@@ -20,6 +20,13 @@ endif()
 
 option(BUILD_SHARED_LIBS "Builds shared libraries instead of static." ${TG_OWT_PACKAGED_BUILD})
 option(TG_OWT_USE_PROTOBUF "Use protobuf to generate additional headers. Useful for packaged build." ${BUILD_SHARED_LIBS})
+option(TG_OWT_USE_PACKAGED_LIBABSL "Build with installed version of abseil-cpp" ${TG_OWT_PACKAGED_BUILD})
+option(TG_OWT_USE_PACKAGED_LIBOPENH264 "Build with installed version of openh264" ${TG_OWT_PACKAGED_BUILD})
+option(TG_OWT_USE_PACKAGED_LIBUSRSCTP "Build with installed version of usrsctp" ${TG_OWT_PACKAGED_BUILD})
+option(TG_OWT_USE_PACKAGED_LIBEVENT "Build with installed version of libevent" ${TG_OWT_PACKAGED_BUILD})
+
+# Currently requires git version (v1.9.0 is latest)
+option(TG_OWT_USE_PACKAGED_LIBVPX "Build with installed version of libvpx" FALSE)
```

Comment:
Please don't do that. tdesktop already had options for each dependency and that was a nightmare. Just make all the libraries search for packaged versions and silently fallback to bundled ones as tdesktop currently does.

Reply:
Alright, I'll change it up, but I'll at least mention why I chose to do it like this. There's plenty of reasons why a packager might choose to use the bundled version of a library, such as the distribution shipping an incompatible version (as is the case with libvpx here, hence disabled by default), and uninstalling the library system-wide just to compile this program being unwieldy. This leads to [hacky `sed` commands](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-im/telegram-desktop/telegram-desktop-2.4.11.ebuild#n100) in package build scripts to replace what would otherwise be achieved by toggles like this, and sometimes inconsistent/unverifiable build results depending on the machine it's built on. To make sure the packager's choices are explicit and not a result of the circumstances, failing the build if a library is unavailable seems sensible.

Reply:
> as is the case with libvpx here

Are you sure? I guess preston just added the submodule without checkouting to a tag, but it still might work with a release version :thinking: 

> This leads to [hacky `sed` commands](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-im/telegram-desktop/telegram-desktop-2.4.11.ebuild#n100) in package build scripts

Probably it's a time to package tdesktop's rlottie as something like `rlottie-tdesktop`.

Reply:
> Are you sure? I guess preston just added the submodule without checkouting to a tag, but it still might work with a release version :thinking:

Well, I'm getting a bunch of errors when building with v1.9.0: https://pastebin.com/raw/EPdb4nh1 . `vpx_rc_funcs_t` was introduced [in october](https://chromium.googlesource.com/webm/libvpx/+/6dba0d0a058c42ace95aee8048eb043c890531e8%5E%21/#F4), and v1.9.0 was released [in july](https://chromium.googlesource.com/webm/libvpx/+/refs/tags/v1.9.0).

I might be missing something but from what I gather this means they're not compatible :/

Reply:
:(

Reply:
pkg_check_modules supports version checking, maybe specify something like v1.10.0-rc2?

Reply:
> This leads to [hacky `sed` commands](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-im/telegram-desktop/telegram-desktop-2.4.11.ebuild#n100)

btw, just looked at the report, it says "does not compile if rlottie is installed". This shouldn't happen and this is why it's automagic in the first place (its meant to build fine with both packaged and bundled, if it doesn't, it's a bug somewhere). lib_lottie even has ifndefs in order to be able to built with upstream rlottie: https://github.com/desktop-app/lib_lottie/search?q=desktop_app_use_packaged_rlottie.